mona_sdk/auth/auth_classes/oidc_auth.py

```diff
@@ -0,0 +1,72 @@
+import requests
+from mona_sdk.auth.auth_utils import get_current_token_by_api_key
+from mona_sdk.auth.auth_globals import EXPIRES_KEY_IN_OIDC, OIDC_ACCESS_TOKEN_KEY
+from mona_sdk.client_exceptions import MonaInitializationException
+from mona_sdk.auth.auth_requests import (
+    BASIC_HEADER,
+    URLENCODED_HEADER,
+    CLIENT_CREDENTIALS_GRANT_TYPE,
+)
+from mona_sdk.auth.auth_classes.base_auth import Base
+
+
+class OidcAuth(Base):
+    def __init__(self, *args, api_key, oidc_scope=None, **kwargs):
+        super().__init__(*args, api_key, **kwargs)
+        self.oidc_scope = oidc_scope
+        self.expires_key = EXPIRES_KEY_IN_OIDC
+
+        # TODO(elie): Support using refresh tokens, as opposed to authenticating again
+        #   in OIDC.
+        self.should_use_refresh_tokens = False
+
+    def _raise_if_missing_params(self):
+        if not self.user_id:
+            raise MonaInitializationException(
+                f"Mona Client is initiated with an auth mode that requires user_id."
+            )
+
+        if (
+            not self.override_rest_api_host
```

Comment:
I feel like this should be a utility function - checking if any backend url is generatable - it can live in the base class as regular function and be called when needed